src/language/githubReleaseInstaller.ts

```diff
@@ -15,18 +15,33 @@ import { DafnyInstaller, getPreferredVersion } from './dafnyInstallation';
 import { configuredVersionToNumeric } from '../ui/dafnyIntegration';
 const ArchiveFileName = 'dafny.zip';
 
-function getDafnyPlatformSuffix(): string {
-  switch(os.type()) {
-  case 'Windows_NT':
-    return 'win';
-  case 'Darwin':
-    if(os.arch() === 'arm64') {
-      return 'osx-11.0';
-    } else {
-      return 'osx-10.14.2';
+function getDafnyPlatformSuffix(version: string): string {
+  // Since every nightly published after this edit will be configured in the post-3.12 fashion, and this script
+  // fetches the latest nightly, it's safe to just condition this on 'nightly' and not 'nightly-date' for a date
+  // after a certain point.
+  const post312 = version.includes('nightly') || configuredVersionToNumeric(version) >= configuredVersionToNumeric('3.13');
+  if(post312) {
+    switch(os.type()) {
+    case 'Windows_NT':
+      return 'windows-2019';
+    case 'Darwin':
+      return 'macos-11';
+    default:
+      return 'ubuntu-20.04';
+    }
```

Comment:
Yeah. We could decide to leave out the versions (in a later change, so we'd have three cases). They're often useful to judge compatibility when downloading manually, but not otherwise necessary.